src/test/java/com/liferay/faces/util/cache/CacheTest.java

```diff
@@ -0,0 +1,319 @@
+
+		int maxCacheCapacity = 10;
+		Cache<String, String> cache3 = cacheFactoryImpl.getCache(defaultInitialCacheCapacity, maxCacheCapacity);
+
+		for (int i = 0; i < maxCacheCapacity; i++) {
+			cache3.putIfAbsent("key" + i, "value" + i);
+		}
+
+		// Verify that get(), put(), and putIfAbsent() all mark the entry as recently used.
+		Assert.assertNotNull(cache3.get("key0"));
+		cache3.putIfAbsent("key1", "value1");
+		cache3.put("key2", "value2");
+		cache3.put("key" + maxCacheCapacity, "value" + maxCacheCapacity);
+		Assert.assertNull(cache3.get("key3"));
+	}
+
+	@Test
+	public void runCacheTest() {
+
+		CacheFactoryImpl cacheFactoryImpl = new CacheFactoryImpl();
+		Cache<String, String> cache = cacheFactoryImpl.getCache();
+		testCache(cache, 1000);
+	}
+
+	@Test
+	public void runConcurrentCacheMaxCapacityLRUTest() throws Exception {
+
+		CacheFactoryImpl cacheFactoryImpl = new CacheFactoryImpl();
+		int defaultInitialCacheCapacity = cacheFactoryImpl.getDefaultInitialCapacity();
+		Cache<String, String> cache1 = cacheFactoryImpl.getConcurrentCache(defaultInitialCacheCapacity, 1000);
+		testCache(cache1, 1000);
+
+		final Cache<String, String> cache2 = cacheFactoryImpl.getConcurrentCache(defaultInitialCacheCapacity, 10);
+		final Queue<Throwable> testErrors = new ConcurrentLinkedQueue<Throwable>();
+		final Queue<AssertionError> testFailures = new ConcurrentLinkedQueue<AssertionError>();
+		testConcurrentCache(cache2, testErrors, testFailures);
+
+		final Cache<String, String> cache3 = cacheFactoryImpl.getConcurrentCache(defaultInitialCacheCapacity, 10);
+		testConcurrentMaxCacheCapacityCache(cache3, 10, testErrors, testFailures);
+	}
+
+	@Test
+	public void runConcurrentCacheTest() throws Exception {
+
+		CacheFactoryImpl cacheFactoryImpl = new CacheFactoryImpl();
+		Cache<String, String> cache1 = cacheFactoryImpl.getConcurrentCache();
+		testCache(cache1, 1000);
+
+		final Cache<String, String> cache2 = cacheFactoryImpl.getConcurrentCache();
+		final Queue<AssertionError> testFailures = new ConcurrentLinkedQueue<AssertionError>();
+		final Queue<Throwable> testErrors = new ConcurrentLinkedQueue<Throwable>();
+		testConcurrentCache(cache2, testErrors, testFailures);
+	}
+
+	private void errorOrFailTestIfNecessary(final Queue<Throwable> testErrors, final Queue<AssertionError> testFailures)
+		throws AssertionError, Exception {
+
+		for (Throwable testError : testErrors) {
+			logger.error("", testError);
+		}
+
+		for (AssertionError testFailure : testFailures) {
+			logger.error("", testFailure);
+		}
+
+		int testErrorsCapacity = testErrors.size();
+
+		if (testErrorsCapacity > 0) {
+			throw new Exception(testErrorsCapacity + " threads threw an error during test execution.");
+		}
+
+		int testFailuresCapacity = testFailures.size();
+
+		if (testFailuresCapacity > 0) {
+			throw new AssertionError(testFailuresCapacity + " threads reported a failure during test execution.");
+		}
+	}
+
+	private void testCache(Cache cache, int iterations) {
```

Comment:
@stiemannkj1 IDE reports a warning that `iterations` is always equal to 1000 which means it doesn't need to be passed as a parameter.